File: backendDotnet/DatabaseSerializer/SerializededModels/Logs/Log.cs
using MongoDB.Bson;

namespace Giger.SerializededModels.Logs
{
    public class Log : Models.Logs.Log
    {
        public Log()
        {
            Id = ObjectId.GenerateNewId().ToString();
            Timestamp = DateTime.Now;
            SourceUserId = "123456";
            SourceUserName = "jsilver";
            SourceHackerName = null;
            TargetUserId = "123457";
            TargetUserName = "adude";
            LogType = Models.Logs.LogType.Transfer;
            LogData = $"Transaction from {SourceUserName} to {TargetUserName} on {Timestamp}";
            SubnetworkId = "SN999";
            SubnetworkName = "Test Subnetwork";
        }
    }
}

File: hacking/src/renderer/services/ApiService/mappers/log.ts
import moment from 'moment';

export default function mapLog(data: any) {
  return {
    data: data.logData,
    type: data.logType,
    sourceUserName: data.sourceUserName,
    sourceUserId: data.sourceUserId,
    sourceHackerName: data.sourceHackerName,
    subnetworkId: data.subnetworkId,
    subnetworkName: data.subnetworkName,
    targetUserId: data.targetUserId,
    targetUserName: data.targetUserName,
    time: moment(data.timestamp).format('DD/MM HH:mm:ss'),
  };
}
